vmm/src/config.rs

```diff
@@ -0,0 +1,224 @@
+    pub fn parse(memory: &str) -> Result<Self> {
+        Ok(MemoryConfig(
+            memory.parse::<u64>().map_err(Error::ParseMemoryParams)?,
+        ))
+    }
+}
+
+impl From<&MemoryConfig> for u64 {
+    fn from(val: &MemoryConfig) -> Self {
+        val.0
+    }
+}
+
+pub struct KernelConfig<'a> {
+    pub path: &'a Path,
+}
+
+impl<'a> KernelConfig<'a> {
+    pub fn parse(kernel: &'a str) -> Result<Self> {
+        Ok(KernelConfig {
+            path: Path::new(kernel),
+        })
+    }
+}
+
+pub struct CmdlineConfig {
+    pub args: Cmdline,
+    pub offset: GuestAddress,
+}
+
+impl CmdlineConfig {
+    pub fn parse(cmdline: Option<&str>) -> Result<Self> {
+        let cmdline_str = cmdline
+            .map(std::string::ToString::to_string)
+            .unwrap_or_else(String::new);
+        let mut args = Cmdline::new(arch::CMDLINE_MAX_SIZE);
+        args.insert_str(cmdline_str).unwrap();
+
+        Ok(CmdlineConfig {
+            args,
+            offset: CMDLINE_OFFSET,
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct DiskConfig<'a> {
+    pub path: &'a Path,
+}
+
+impl<'a> DiskConfig<'a> {
+    pub fn parse(disk: &'a str) -> Result<Self> {
+        Ok(DiskConfig {
+            path: Path::new(disk),
+        })
+    }
+}
+
+pub struct RngConfig<'a> {
+    pub src: &'a Path,
+}
+
+impl<'a> RngConfig<'a> {
+    pub fn parse(rng: Option<&'a str>) -> Result<Option<Self>> {
+        if rng.is_none() {
+            return Ok(None);
+        }
+
+        Ok(Some(RngConfig {
+            src: Path::new(rng.unwrap()),
+        }))
+    }
+}
+
+pub struct NetConfig<'a> {
+    pub tap: Option<&'a str>,
+    pub ip: Ipv4Addr,
+    pub mask: Ipv4Addr,
+    pub mac: MacAddr,
+}
+
+impl<'a> NetConfig<'a> {
+    pub fn parse(net: Option<&'a str>) -> Result<Option<Self>> {
+        if net.is_none() {
+            return Ok(None);
+        }
+
+        // Split the parameters based on the comma delimiter
+        let params_list: Vec<&str> = net.unwrap().split(',').collect();
+
+        let mut tap_str: &str = "";
+        let mut ip_str: &str = "";
+        let mut mask_str: &str = "";
+        let mut mac_str: &str = "";
+
+        for param in params_list.iter() {
+            if param.starts_with("tap=") {
+                tap_str = &param[4..];
+            } else if param.starts_with("ip=") {
+                ip_str = &param[3..];
+            } else if param.starts_with("mask=") {
+                mask_str = &param[5..];
+            } else if param.starts_with("mac=") {
+                mac_str = &param[4..];
+            }
+        }
+
+        let mut tap: Option<&str> = None;
+        let mut ip: Ipv4Addr = "192.168.249.1".parse().unwrap();
+        let mut mask: Ipv4Addr = "255.255.255.0".parse().unwrap();
+        let mut mac: MacAddr = MacAddr::local_random();
+
+        if !tap_str.is_empty() {
+            tap = Some(tap_str);
+        }
+        if !ip_str.is_empty() {
+            ip = ip_str.parse().unwrap();
```

Comment:
Need to `map_err (...)?` here and for mask and MAC.

Reply:
Done!